common/txmgr/types/storage_service.go

```diff
@@ -6,30 +6,33 @@ import (
 
 	uuid "github.com/satori/go.uuid"
 
+	"github.com/smartcontractkit/chainlink/v2/common/types"
 	"github.com/smartcontractkit/chainlink/v2/core/services/pg"
 )
 
 // NEWTX, TX, TXATTEMPT will be converted from generic types to structs at a future date to enforce design and type checks
-type TxStorageService[ADDR any, CHAINID any, HASH any, NEWTX any, R any, TX any, TXATTEMPT any, TXID any, TXMETA any] interface {
+//
+//go:generate mockery --quiet --name TxStorageService --output ./mocks/ --case=underscore
+type TxStorageService[ADDR any, CHAINID any, TX_HASH types.Hashable, BLOCK_HASH types.Hashable, NEWTX any, R any, TX any, TXATTEMPT any, TXID any, TXMETA any] interface {
 	UnstartedTxQueuePruner
 	CheckEthTxQueueCapacity(fromAddress ADDR, maxQueuedTransactions uint64, chainID CHAINID, qopts ...pg.QOpt) (err error)
 	CountUnconfirmedTransactions(fromAddress ADDR, chainID CHAINID, qopts ...pg.QOpt) (count uint32, err error)
 	CountUnstartedTransactions(fromAddress ADDR, chainID CHAINID, qopts ...pg.QOpt) (count uint32, err error)
-	CreateEthTransaction(newTx NEWTX, chainID CHAINID, qopts ...pg.QOpt) (tx TX, err error)
+	CreateEthTransaction(newTx NEWTX, chainID CHAINID, qopts ...pg.QOpt) (tx Transaction, err error)
```

Comment:
Yes, it should actually be the EthTx type, which is already a generic type "type EthTx[ADDR commontypes.Hashable, TX_HASH commontypes.Hashable]".

However, currently EthTx is defined in txmgr package, and that package depends on this common/txmgr/types package.
I couldn't use that because of cyclic deps. So left it or a follow-up PR.
We need to move EthTxAttempt, EthTxMeta and EthTx to this package after removing all EVM specific references, and rename them to just TxAttempt, Tx, TxMeta. That is something I didn't want to tackle in this PR.

Reply:
ok, that makes sense - can you leave a TODO comment so we know that still needs to be done?

Reply:
done